docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,121 @@
+# Application dependencies
+
+Starting from Argo CD v2.9, you can natively model dependencies between Applications. This new mechanism is intended to replace the prior way of modeling dependencies when using the app-of-apps pattern with sync waves on the Application custom resources.
+
+Using the new dependency mechanism, users are not bound to the app-of-apps pattern anymore and can define dependencies between applications using application selectors to define a directed, acyclic graph for dependency modeling. This means that an Application which defines one or more dependencies do not only depend on its direct dependencies, but to the dependencies of its dependencies as well.
+
+Application dependencies support both, manual and automatic sync policies.
+
+## Feature state and caveats
+
+This feature is in [beta state](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md#beta). It will be polished across the upcoming releases.
+
+Bug reports and enhancement requests are most welcome.
+
+## Limitations and known issues
+
+A list of things that are good to know before getting started with Application dependencies:
+
+* An Application can only depend on other Applications that use the same AppProject and are in the same namespace as the dependent Application.
+* A circular dependency defined anywhere in the dependency chain is considered to be an error and will prevent any application in the chain from syncing.
+* Syncing an application that is dependent on other apps will not trigger a sync of the dependencies. However, manual sync of a dependent application whose dependencies are not satisfied will block.
```

Comment:
What is the definition of a satisfied dependency vs not satisfied. I think we need to clarify it in the doc. 
If service A depends on service B and B is in degraded state, what will happen when trying to sync A? What are all conditions that would block Service A to sync?

Reply:
Agree, fair call. In short, the conditions are the same as for sync waves (i.e. application's sync status must be `Synced` and health status be `Healthy`). I'll make sure to write that out in the docs.

Reply:
I updated the docs. If it still doesn't make sense what I wrote (I have the tendency to assume too much implicit knowledge by the user), please feel free to call that out :)

Reply:
Thank you. Added a small suggestion but it looks good now!